editorconfig.el

```diff
@@ -378,6 +378,32 @@ TRIM-TRAILING-WS."
              (> (string-to-number length) 0))
     (setq fill-column (string-to-number length))))
 
+(defun editorconfig--is-a-mode-p (current want)
+  "Return non-nil if major mode CURRENT is a major mode WANT."
+  (or (eq current
+          want)
+      (let ((parent (get current 'derived-mode-parent)))
+        (and parent
+             (editorconfig--is-a-mode-p parent want)))))
+
+(defun editorconfig-set-major-mode (filetype)
+  "Set buffer `major-mode' by FILETYPE.
+
+FILETYPE should be s string like `\"ini\"`, if not nil or empty string."
+  (let ((mode (and filetype
+                   (not (string= filetype
+                                 ""))
+                   (intern (concat filetype
+                                   "-mode")))))
```

Comment:
I don't know too much about this, but it looks like major modes are just functions. I [found this example major mode](https://www.emacswiki.org/emacs/wpdl-mode.el) to show how it's defined with a defun. 

I guess maybe it is an implicit requirement to end modes with `-mode` since otherwise there's no way to destinguish them.

Maybe you could look for `*-hook`, but that seems more hacky than the current way...